airflow/providers/google/cloud/triggers/vertex_ai.py

```diff
@@ -84,16 +97,66 @@ async def run(self) -> AsyncIterator[TriggerEvent]:
             return
```

Comment:
Maybe there is a point in making all trigger events, including those with exceptions, follow the same pattern of having a status, message and job? 

Also we can probably make the trigger event with an exception more informative, if we include the job id / name into the message.

What do you think?

Reply:
> Maybe there is a point in making all trigger events, including those with exceptions, follow the same pattern of having a status, message and job?

If the exception occurred during the job wait, probably the job is corrupted, and in this case we would have to catch another exception during the call `self._serialize_job(job)`. In my opinion, it wouldn't look good and simple, thus I avoided doing that. But if it seems reasonable for the team, I will certainly do it.

> Also we can probably make the trigger event with an exception more informative, if we include the job id / name into the message.

I agree and have nothing against that. I will add it in the following commit with the other fixes. Thanks for the suggestion!

Reply:
Fixed. Please, take a look.

Reply:
Looks great